lib/api/utilities/functions.php

```diff
@@ -81,172 +81,173 @@ function beans_remove_dir( $dir_path ) {
 	unset( $items[0], $items[1] );
 
 	foreach ( $items as $needle => $item ) {
-
 		$path = $dir_path . '/' . $item;
 
-		if ( 'dir' === filetype( $dir_path . '/' . $item ) ) {
+		if ( is_dir( $path ) ) {
 			beans_remove_dir( $path );
 		} else {
 			@unlink( $path );
 		}
-
-		unset( $items[ $needle ] );
-
 	}
 
-	@rmdir( $dir_path );
-
-	return true;
-
+	return @rmdir( $dir_path );
 }
 
 /**
  * Convert internal path to a url.
  *
  * This function must only be used with internal paths.
  *
- * @since 1.0.0
+ * @since 1.5.0
  *
- * @param string $path Path to be converted. Accepts absolute and relative internal paths.
+ * @param string $path          Path to be converted. Accepts absolute and relative internal paths.
+ * @param bool   $force_rebuild (Optional) Forces the rebuild of the root url and path.
  *
  * @return string Url.
  */
-function beans_path_to_url( $path ) {
-
-	static $root, $host;
+function beans_path_to_url( $path, $force_rebuild = false ) {
```

Comment:
For testing purposes, I added a `force_rebuild` parameter.  There may also be times where you want to force a rebuild.  One use case that comes to mind is:

In a multisite, I've had situations where I need to programmatically switch to a different site, grab some information and process, and then switch back.  In this case, I would need the ability to clear the cache and rebuild.